### ISSUE TITLE
First loadedSource event raised by Debug Adapter adds nothing to the Loaded Scripts panel

### ISSUE THREAD

**Issue by gjsjohnmurray** [Microsoft/vscode#59281]
Version: 1.28.0-insider (user setup)
Commit: 79f79a2c971d6eab36b92d78505402b42ffcae93
Date: 2018-09-24T05:19:32.981Z
Electron: 2.0.9
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

Steps to Reproduce:

1. Code a Debug Adapter capable of raising loadedSource events, e.g. via evaluateRequest's handling of the 'repl' context. Mine was derived from the mock-debug sample.
2. The first loadedSource event you raise with reason='new' produces no visible entry in the Loaded Scripts panel.
3. The second loadedSource event (different Source) results in two entries appearing.
4. Thereafter, each loadedSource event correctly adds its Source to the panel.

**Comment by isidorn** [comment]
@gjsjohnmurray are you sure you are sending your first loadedSource event after the initalisation has been done and you have responsed properly to the initialize request?
Also not to refresh the tree too much there is 300 ms timeout. You still do not see the tree updating after this time?

**Comment by gjsjohnmurray** [comment]
@isidorn yes, pretty sure I'm not making those mistakes.

To replicate it in the vscode-mock-debug sample I did the following:

1. Added this to initializeRequest
		response.body.supportsLoadedSourcesRequest = true;

2. Modified evaluateRequest to support REPL strings such as 'ls 1' and 'ls 2'
		if (args.context === 'repl') {
			// 'evaluate' supports adding fake entries to Loaded Scripts panel via the 'repl':

			const matches = /ls +([0-9]+)/.exec(args.expression);
			if (matches && matches.length === 2) {
				const src = new Source(`YJM${matches[1]}.txt`, `debug:YJM${matches[1]}.txt`);
				this.sendEvent(new LoadedSourceEvent('new', src));
				reply = `entry added to Loaded Scripts panel`;
			}
		}

3. Ran the amended vscode-mock-debug in the VSCode debugger as usual. When debugging of the readme.md file was underway, entered the following commands into the input field of the Debug Output panel:

>ls 1

Debug Output echoed the command, followed by:
entry added to Loaded Scripts panel

But nothing appeared in the Loaded Scripts panel.

![image](https://user-images.githubusercontent.com/6726799/46009662-1b81d400-c075-11e8-96df-98e9d4db55ff.png)

>ls 2

Again, this echoed, followed by:
entry added to Loaded Scripts panel

At this point both entries appeared:
![image](https://user-images.githubusercontent.com/6726799/46009719-3d7b5680-c075-11e8-94c6-405d3946c5b6.png)
